plotly/io/_json.py

```
@@ -0,0 +1,193 @@
+    if pretty:
+        opts['indent'] = 2
+    else:
+        # Remove all whitespace
+        opts['separators'] = (',', ':')
+
+    return json.dumps(fig_dict, cls=PlotlyJSONEncoder, **opts)
+
+
+def write_json(fig, file, validate=True, pretty=False, remove_uids=True):
+    """
+    Convert a figure to JSON and write it to a file or writeable
+    object
+
+    Parameters
+    ----------
+    fig:
+        Figure object or dict representing a figure
+
+    file: str or writeable
+        A string representing a local file path or a writeable object
+        (e.g. an open file descriptor)
+
+    pretty: bool (default False)
+        True if JSON representation should be pretty-printed, False if
+        representation should be as compact as possible.
+
+    remove_uids: bool (default True)
+        True if trace UIDs should be omitted from the JSON representation
+
+    Returns
+    -------
+    None
+    """
+
+    # Get JSON string
+    # ---------------
+    # Pass through validate argument and let to_json handle validation logic
+    json_str = to_json(
+        fig, validate=validate, pretty=pretty, remove_uids=remove_uids)
+
+    # Check if file is a string
+    # -------------------------
+    file_is_str = isinstance(file, string_types)
+
+    # Open file
+    # ---------
+    if file_is_str:
+        with open(file, 'w') as f:
+            f.write(json_str)
+    else:
+        file.write(json_str)
+
+
+def from_json(value, output_type='Figure', skip_invalid=False):
+    """
+    Construct a figure from a JSON string
+
+    Parameters
+    ----------
+    value: str
+        String containing the JSON representation of a figure
+    output_type: type or str (default 'Figure')
+        The output figure type or type name.
+        One of:  graph_objs.Figure, 'Figure,
+                 graph_objs.FigureWidget, 'FigureWidget'
+    skip_invalid: bool (default False)
+        False if invalid figure properties should result in an exception
```

and "True if invalid figure properties should be silently ignored."